src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -347,12 +347,19 @@ protected function instantiateObject(array &$data, string $class, array &$contex
             }
 
             $constructorParameters = $constructor->getParameters();
+            $attributesMetadata = $this->classMetadataFactory ? $this->classMetadataFactory->getMetadataFor($class)->getAttributesMetadata() : null;
 
             $params = [];
             foreach ($constructorParameters as $constructorParameter) {
                 $paramName = $constructorParameter->name;
                 $key = $this->nameConverter ? $this->nameConverter->normalize($paramName, $class, $format, $context) : $paramName;
 
+                $attributeContext = [];
+
+                if ($attributesMetadata && \array_key_exists($paramName, $attributesMetadata)) {
```

Comment:
Why using `array_key_exists` here over a simple `isset` ? This would fail on `$attributesMetadata[$paramName]` being `null` (which should not happen, but `array_key_exists` is usually used when differentiating `null` matters)

```suggestion
                if (isset($attributesMetadata[$paramName])) {
```

Reply:
Well, i did it with `array_key_exists` since `isset` is doing to much imho. But can change it, if wanted.